qemu/tests/virtio_console.py

```diff
@@ -1845,24 +1826,10 @@ def test_delete_guest_script():
     #
     fce = None
     _fce = "test_" + params.get('virtio_console_test', '').strip()
-    error.context("Executing test: %s" % _fce, logging.info)
+    error_context.context("Executing test: %s" % _fce, logging.info)
     if _fce not in locals():
-        raise error.TestNAError("Test %s doesn't exist. Check 'virtio_console_"
-                                "test' variable in subtest.cfg" % _fce)
+        test.cancel("Test %s doesn't exist. Check 'virtio_console_"
+                    "test' variable in subtest.cfg" % _fce)
     else:
-        try:
-            fce = locals()[_fce]
-            return fce()
-        except Exception, details:
-            EXIT_EVENT.set()    # Avoid hangs
-            exc_type, exc_value, exc_traceback = sys.exc_info()
-            logging.error("Original traceback:\n" +
-                          "".join(traceback.format_exception(
-                                  exc_type, exc_value,
-                                  exc_traceback.tb_next)))
-            if isinstance(details, error.TestError):
-                raise error.TestError('%s error: %s' % (_fce, details))
-            elif isinstance(details, error.TestNAError):
-                raise error.TestNAError('%s skipped: %s' % (_fce, details))
-            else:
-                raise error.TestFail('%s failed: %s' % (_fce, details))
+        fce = locals()[_fce]
+        return fce()
```

Comment:
why do you delete the `excpt` part?

Reply:
Thanks. This code block applied `isinstance` to determine the exception type, which needs to import avocado.core module, so the reason is to avoid this importing in special test script.

Reply:
Seems we should keep `EXIT_EVENT.set()`.

Reply:
@luckyh Thank for advice, I modified my code, could you help review it again? Thanks!

RHEL7:
smp_2.1024m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.myprovider.virtio_console.spread_linear.incorrect_options.boot_too_much_ports: PASS (6.05 s)
RHEV7:
smp_2.1024m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.myprovider.virtio_console.spread_linear.incorrect_options.boot_too_much_ports: PASS (6.09 s)